common/storage.go

```diff
@@ -0,0 +1,177 @@
+package common
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/giantswarm/cluster-test-suites/assets/storage"
+	"github.com/giantswarm/cluster-test-suites/helper"
+	"github.com/giantswarm/clustertest/pkg/client"
+	"github.com/giantswarm/clustertest/pkg/wait"
+	corev1 "k8s.io/api/core/v1"
+	storagev1 "k8s.io/api/storage/v1"
+	apierror "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	cr "sigs.k8s.io/controller-runtime/pkg/client"
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+)
+
+var (
+	namespace = "test-storage"
+)
+
+func runStorage() {
+	Context("storage", func() {
+		var wcClient *client.Client
+
+		BeforeEach(func() {
+			var err error
+
+			wcClient, err = Framework.WC(Cluster.Name)
+			if err != nil {
+				Fail(err.Error())
+			}
+		})
+
+		It("has a at least one storage class available", func() {
+			Eventually(wait.Consistent(checkStorageClassExists(wcClient), 10, time.Second)).
+				WithTimeout(wait.DefaultTimeout).
+				WithPolling(wait.DefaultInterval).
+				Should(Succeed())
+		})
+
+		When("a pod uses a persistent volume claim", func() {
+			BeforeEach(func() {
+				Eventually(wait.Consistent(createPodWithPVC(wcClient), 10, time.Second)).
+					WithTimeout(wait.DefaultTimeout).
+					WithPolling(wait.DefaultInterval).
+					Should(Succeed())
+			})
+
+			AfterEach(func() {
+				Eventually(wait.Consistent(deleteStorage(wcClient, namespace), 10, time.Second)).
+					WithTimeout(wait.DefaultTimeout).
+					WithPolling(wait.DefaultInterval).
+					Should(Succeed())
+			})
+
+			It("runs successfully", func() {
+				Eventually(wait.Consistent(verifyPodState(wcClient, "pvc-test-pod", namespace), 10, time.Second)).
+					WithTimeout(wait.DefaultTimeout).
+					WithPolling(wait.DefaultInterval).
+					Should(Succeed())
+			})
+		})
+	})
+}
+
+func cleanupStorage() {
+	Context("storage", func() {
+		var wcClient *client.Client
+
+		BeforeEach(func() {
+			var err error
+
+			wcClient, err = Framework.WC(Cluster.Name)
+			if err != nil {
+				Fail(err.Error())
+			}
+		})
+
+		It("has deleted all objects in namespace test-storage", func() {
+			Eventually(wait.Consistent(deleteStorage(wcClient, namespace), 10, time.Second)).
+				WithTimeout(wait.DefaultTimeout).
+				WithPolling(wait.DefaultInterval).
+				Should(Succeed())
+		})
+	})
+}
```

Comment:
Is this doing anything that's not already covered on line 54?

Reply:
Yeah you're right, removed it. Thanks <3